### ISSUE TITLE
Builder status stuck on first commit it has built.

### ISSUE THREAD

**Issue by mihneadb** [janinko/ghprb#263]
Steps to reproduce:

* open PR
* trigger build using passphrase
  * build is executed & status is updated for **commit1**
* make & push new commit (**commit2**)
* trigger build using passphrase
  * build is executed on **commit2**, but status is updated for **commit1**

**Comment by mihneadb** [comment]
This happens with ghprb:1.20 and github-api:1.67. Things work fine with ghprb:1.17 and github-api:1.66

**Comment by mihneadb** [comment]
Wrong - things work fine after restarting jenkins but after you push a commit and trigger another test it happens again. @janinko, have you seen this before? Thanks!

**Comment by mihneadb** [comment]
I've investigated a bit, it seems that the PR's `head` is not updated.

**Comment by mihneadb** [comment]
Found other reports of (probably) the same issue:

* https://github.com/janinko/ghprb/issues/150
* https://github.com/janinko/ghprb/issues/43
* https://github.com/janinko/ghprb/issues/19

**Comment by DavidTanner** [comment]
Other than checking if the project is disabled, there haven't been any changes to GhprbPullRequest where the update logic exists.  I also can't recreate it locally.

Are you using webhooks?

**Comment by DavidTanner** [comment]
Here is my testing repo https://github.com/DavidTanner/Testing/pull/14

**Comment by mihneadb** [comment]
@DavidTanner Yes, I am using webhooks + build phrase (not auto builds like in your test). Could you try to reproduce it that way? By reading through the code it seems that the object is instantiated once with the latest HEAD and it never gets updated (at least via webhook build phrase) -- or maybe I misread, but that's the behavior.

**Comment by DavidTanner** [comment]
Maybe the changes I have been making have fixed things, but it works with webhooks as well.

**Comment by mihneadb** [comment]
Are you planning a new release? Would love to try it out :).

**Comment by DavidTanner** [comment]
1.22.2 is out there, and that is what I was just testing with.

**Comment by mihneadb** [comment]
I've just tested with 1.22.2 and I encounter the same behavior. :( Triggering the build phrase after the 2nd commit merges the correct sha (latest version of the code), but updates the status of the first commit.

If you have any suggestions of what to try, pls let me know.

**Comment by DavidTanner** [comment]
The next step is to setup a logger, then if that doesn't give us the data we need then i can send you a rogue build with more logging. 
Go to Manage > System Log and add a logger for

org.jenkinsci.plugins.ghprb

That will log everything from the plugin.   
After the commit you should see:

Pull request #X was updated ...
New Commit. Sha: ...

That should happen before the comment gets sent.

**Comment by mihneadb** [comment]
Here are the logs. As you can see, the "pull req was updated" part does not show up. :(

https://gist.github.com/mihneadb/2d6ce82981f22168ff54

**Comment by DavidTanner** [comment]
Can you please retest with 1.24.8?

**Comment by mihneadb** [comment]
It's a bit difficult now since we've changed our setup, but I'll get back with the results whenever I get to it.
